### DISCUSSION TITLE
About connect() implementation

### DISCUSSION THREAD

**Discussion by aaazzz1234** [ANLAB-KAIST/KENSv3#172] [Project Q&A 2. Transport Layer (TCP)] [answered]
Hello, I'm now implementing connect() sys-call with 3-way handshaking.
while doing it, I thought connect() function should send packet to server.
To construct packet, it needs ip-address and port of client and server and etc.....
In case of server's ip and port, it is given by argument 2.
But how can i find client's IP address?? (In case of port, it can be set arbitrary)

**Comment by Pusnow** [comment, marked as answer]
You should routing table. See: https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Figuring-out-the-source-IP-address-to-use
